casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala

```diff
@@ -23,15 +31,29 @@ final case class Genesis(
 )
 
 object Genesis {
+  val genesisPubKey = PublicKey(Array[Byte]())
```

Comment:
We need to create concise logic how random seed is created in different situations.

For this purpose we created specialized type `BlockRandomSeed` so it make sense to put the logic there.

`genesisPubKey`:
1. is not really genesis public key so it's misleading
2. is not used for any other purpose that to be part of data to create a seed so no need to be available outside of the scope where it's used.

On many places where seed is needed the same code is duplicated and it should be moved to `BlockRandomSeed`. Create helper functions for each scenario which will serve as a reference to show how many cases we have to handle.

Don't try to create everything in one expression which then becomes stretched through many lines.
```scala
val seed = BlockRandomSeed(shardId, 0, genesisPubKey, emptyStateHashFixed.toBlake2b256Hash)
val rand = generateSplitRandomNumber(seed, nonNegativeContractIndex, splitIdx)
```